src/plugins/data/server/index_patterns/deprications/scripted_fields.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import {
+  CoreSetup,
+  DeprecationsDetails,
+  GetDeprecationsContext,
+  RegisterDeprecationsConfig,
+} from 'kibana/server';
+import { IndexPatternAttributes } from '../../../common';
+
+type IndexPatternAttributesWithFields = Pick<IndexPatternAttributes, 'title' | 'fields'>;
+
+export const createScriptedFieldsDeprecationsConfig: (
+  core: CoreSetup
+) => RegisterDeprecationsConfig = (core: CoreSetup) => ({
+  getDeprecations: async (context: GetDeprecationsContext): Promise<DeprecationsDetails[]> => {
+    const finder = context.savedObjectsClient.createPointInTimeFinder<IndexPatternAttributesWithFields>(
+      {
+        type: 'index-pattern',
+        perPage: 100,
+        fields: ['title', 'fields'],
+      }
+    );
+
+    let indexPatternsWithScriptedFields: IndexPatternAttributesWithFields[] = [];
+    for await (const response of finder.find()) {
+      indexPatternsWithScriptedFields = response.saved_objects
+        .map((so) => so.attributes)
+        .filter(hasScriptedField);
+
+      if (indexPatternsWithScriptedFields.length > 0) {
+        // no need to look up all index patterns since we've found at least one that has scripted fields
+        await finder.close();
+        break;
```

Comment:
I intentionally don't fetch all index patterns to make this function cheaper. This is reflected in the wording of the message below:  `You have **at least** <n> index patterns...`

Reply:
I think we should show a full list of index patterns that need updating as some setups may have a large number of index patterns and only a few might have scripted fields. We need a way to deliminate the index patterns as they can have commas in them. Could we link directly to the index patterns?

Reply:
> Could we link directly to the index patterns?

Nope, as I understand we can only use plain text.

At this moment this pr shows at max 3 index pattern titles as a hint. 

1. Do you think we should show more out from the page? (e.g. in the current implementation we can remove that `.slice(0,3)` where I prepare the message
2. Or do you think we should fetch all the index patterns and show the whole list no matter how long it is?

In the current implementation user can: 

1. Take a look at a message, see up to 3 index pattern titles that need to be fixed
2. Fix them 
3. If there are more to fix, see another 3 index pattern titles and fix them 

Not ideal, but not sure we want to fetch all the index patterns?

Reply:
I update so in the main message we show just up to 3 index pattern titles, but then when you open dialog with details we list all of them!